packages/graphql-codegen-cli/src/codegen.ts

```diff
@@ -257,11 +257,17 @@ export async function executeCodegen(input: CodegenContext | Types.Config): Prom
                     task: wrapTask(async () => {
                       debugLog(`[CLI] Loading Documents`);
 
-                      const allDocuments = [...rootDocuments, ...outputSpecificDocuments];
+                      // get different cache for shared docs and output specific docs
+                      const results = await Promise.all(
+                        [rootDocuments, outputSpecificDocuments].map(docs => {
+                          const hash = JSON.stringify(docs);
+                          return documentsLoadingCache.load(hash);
+                        })
+                      );
+
+                      const documents: Types.DocumentFile[] = [];
 
-                      const hash = JSON.stringify(allDocuments);
-                      const result = await documentsLoadingCache.load(hash);
-                      const documents: Types.DocumentFile[] = result.documents;
+                      results.forEach(source => documents.push(...source.documents));
```

Comment:
Won't this break if you have identical operations with near-operation-file preset?

Reply:
Discussed offine 👍🏼